wfe2/wfe.go

```diff
@@ -2353,14 +2354,110 @@ func (wfe *WebFrontEndImpl) FinalizeOrder(ctx context.Context, logEvent *web.Req
 	}
 }
 
-// certID matches the ASN.1 structure of the CertID sequence defined by RFC6960.
-type certID struct {
+// ariCertID is an interface that represents a unique identifier for a
+// certificate. It is implemented by both the deprecatedCertID and certID
+// structs.
+//
+// TODO(#7183): Remove this.
+type ariCertID interface {
+	Serial() string
+}
+
+// deprecatedCertID matches the ASN.1 structure of the CertID sequence defined
+// by RFC6960. Its fields are exported so that they can be unmarshaled using the
+// asn1 package.
+//
+// TODO(#7183): Remove this.
+type deprecatedCertID struct {
 	HashAlgorithm  pkix.AlgorithmIdentifier
 	IssuerNameHash []byte
 	IssuerKeyHash  []byte
 	SerialNumber   *big.Int
 }
 
+func (c deprecatedCertID) Serial() string {
+	return core.SerialToString(c.SerialNumber)
+}
+
+// parseDeprecatedCertID parses a unique identifier (certID) for a certificate
+// as per the ACME protocol's "renewalInfo" resource, as specified in
+// draft-ietf- acme-ari-02. For more details see:
+// https://datatracker.ietf.org/doc/html/draft-ietf-acme-ari-01#section-4.1.
+//
+// TODO(#7183): Remove this.
+func parseDeprecatedCertID(path string) (ariCertID, error) {
+	der, err := base64.RawURLEncoding.DecodeString(path)
+	if err != nil {
+		return deprecatedCertID{}, err
```

Comment:
```suggestion
		return nil, err
```

Since this function returns `ariCertID` (an interface type) it can return nil.

Reply:
🤩 that's beautiful.